### PR TITLE
Fixes AI camera static not updating when the AI enters a mecha

### PR DESCRIPTION
**Pull request by Sealed101**
## About The Pull Request
AI camera static updates when `setLoc` is called, moving the AI eye to a new turf. Upon entering a mecha, it was only called in resetting the AI eye's loc to the AI core. 
Thus, any static images that were present are now left over wherever the AI eye was when the AI had entered the mech, and they won't move according to the movement of the mecha. That made it possible to exit the area where the static was set and cause the mecha AI to have unobstructed vision. 
This PR makes it so the AI eye is instead moved to the mecha along with the AI. While in the mecha, the static will update via a signal registered on mecha movement. The cameranet's `visibility` proc was changed to use the AI eye's turf and its coordinates to calculate the area wrapped in static.

## Why It's Good For The Game
Prevents use of mecha with completely unobstructed vision in areas where there should be static, which is basically an exploit.

## Changelog
:cl:
fix: Fixed AIs piloting a mecha gaining unobstructed vision with no cameras present when moving past a certain point
/:cl:

### CODE REVIEW

**Comment by ShizCalev** [comment]
#70329

File: code/modules/mob/living/silicon/ai/freelook/cameranet.dm [hunk 1]
```diff
@@ -66,17 +66,19 @@ GLOBAL_DATUM_INIT(cameranet, /datum/cameranet, new)
...
-			var/y2 = min(world.maxy, eye.y + static_range) & ~(CHUNK_SIZE - 1)
+			var/x1 = max(0, eye_turf.x - static_range) & ~(CHUNK_SIZE - 1)
+			var/y1 = max(0, eye_turf.y - static_range) & ~(CHUNK_SIZE - 1)
+			var/x2 = min(world.maxx, eye_turf.x + static_range) & ~(CHUNK_SIZE - 1)
+			var/y2 = min(world.maxy, eye_turf.y + static_range) & ~(CHUNK_SIZE - 1)
 
 			for(var/x = x1; x <= x2; x += CHUNK_SIZE)
 				for(var/y = y1; y <= y2; y += CHUNK_SIZE)
-					visibleChunks |= getCameraChunk(x, y, eye.z)
+					visibleChunks |= getCameraChunk(x, y, eye_turf.z)
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/ai/freelook/cameranet.dm:81]: This is still runtiming.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by Sealed101** [comment]
the camera eye is fine while it is approaching a distance of 32 tiles before the map edge. stepping over to a tile that's 31 tiles shy of the map edge triggers the runtime. relevant code: 
https://github.com/tgstation/tgstation/blob/bb4cce72e8d0d2121e8f8d084619435f2f50c920/code/modules/mob/living/silicon/ai/freelook/cameranet.dm#L67-L79

in the case of the eye reaching a coordinate of 0, the eye's coordinate has `CHUNK_SIZE = 16` substracted from it, then the result filters out in the right hand part of the quation.

icebox and tram have the AI sat close enough to the edge that they constantly runtime for as long as the AI keeps its eye in the sat.

thing is, cameranet creating chunks with some turf coordinates set as 0 is as old as time, so i am not sure how did that not cause any issues. just suppressing the 0th coordinate turfs in the `lowest` won't do, so the solution lies in making camera chunks respect map edges.

the fun thing is, on local testing this breaks static generation on the left and lower edges of the map, the upper and the right edges generate static just fine

**Comment by Sealed101** [comment]
waiting on #70235

**Pull request merged**